src/js/medium-editor.js

```diff
@@ -1220,6 +1220,11 @@ else if (typeof define === 'function' && define.amd) {
 
                     self.createLink(this, target, button);
                 }
+                else if (e.keyCode === 27) {
+                    e.preventDefault();
+                    self.showToolbarActions();
+                    restoreSelection.call(self, self.savedSelection);
```

Comment:
It seems like we would want ESC to behave similar to ENTER/RETURN except that it would actually create the link.  If that's the case, would it be a good idea to factor out the part of the `createLink` method that hides the anchorForm, shows the toolbar, and manipulates the selection into a separate function and just call that from here? This would make ESC go down the same code path as ENTER, just without actually embedding/updating the link.